src/components/Dialogs/AdvancedDelegateDialog/SubdelegationRow.tsx

```diff
@@ -40,6 +42,59 @@ function SubdelegationToRow({
       ? 0
       : Math.round((allowance / availableBalanceNumber) * 100_00) / 100;
 
+  const handleAllowanceInput = (e: React.ChangeEvent<HTMLInputElement>) => {
+    function formatNumber(value: number) {
+      return Math.floor(Math.round(value * 10000) / 10) / 1000;
+    }
+
+    function getCleanInput(value: string) {
```

Comment:
Might be better to pull the functions out the const definition or they will keep getting defined each time.

Reply:
+1

left them inside because `formatNumber` was inside already